### PR TITLE
Log process startup and running time

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

This helps to get insights about slow startup of Loki, e.g. for the index gateway, which may pre-fetch a lot of data before it becomes ready.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: pkg/loki/loki.go [hunk 1]
```diff
@@ -399,6 +399,8 @@ func (t *Loki) ListTargets() {
 
 // Run starts Loki running, and blocks until a Loki stops.
 func (t *Loki) Run(opts RunOpts) error {
```
Comment (chaudum) [inline pkg/loki/loki.go:401]: If we want a more precise startup time, including some config parsing that happens prior to calling `Run()` we could start the timer in the `main()` function and pass it as a `RunOpt`.
Not sure if it really makes much difference, though.

Comment (chaudum) [reply pkg/loki/loki.go:401]: Actually, let me do that!

**Review by ashwanthgoli — APPROVED** [review]
lgtm

File: pkg/loki/loki.go [hunk 2]
```diff
@@ -369,6 +368,9 @@ type RunOpts struct {
 	// CustomConfigEndpointHandlerFn is the handlerFunc to be used by the /config endpoint.
 	// If empty, default handlerFunc will be used.
 	CustomConfigEndpointHandlerFn func(http.ResponseWriter, *http.Request)
+	// StartTime is the time at which the main() function started executing.
+	// It is used to determin the startup time as well as the running time of the Loki process.
```
Comment (ashwanthgoli) [inline pkg/loki/loki.go:372]:
```suggestion
	// It is used to determine the startup time as well as the running time of the Loki process.
```

**Pull request merged**